libs/community/langchain_community/chat_message_histories/falkordb.py

```diff
@@ -0,0 +1,218 @@
+            self._ssl = ssl
+
+            try:
+                self._driver = falkordb.FalkorDB(
+                    host=self._host,
+                    port=self._port,
+                    username=self._username,
+                    password=self._password,
+                    ssl=self._ssl,
+                )
+            except redis.exceptions.ConnectionError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the host and port are correct."
+                )
+            except redis.exceptions.AuthenticationError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the username and password are correct."
+                )
+
+        self._database = self._driver.select_graph(session_id)
+        self._session_id = session_id
+        self._node_label = node_label
+        self._window = window
+
+        self._database.query(
+            f"MERGE (s:{self._node_label} {{id:$session_id}})",
+            {"session_id": self._session_id},
+        )
+
+        try:
+            self._database.create_node_vector_index(
+                f"{self._node_label}", "id", dim=5, similarity_function="cosine"
+            )
+        except Exception as e:
+            if "already indexed" in e:
+                raise ValueError(f"{self._node_label} has already been indexed")
+
+    def _process_records(self, records: list) -> list:
+        """Process the records from FalkorDB and convert them into BaseMessage objects.
+
+        Args:
+            records (list): The raw records fetched from the FalkorDB query.
+
+        Returns:
+            list: A list of `BaseMessage` objects.
+        """
+        messages = []
+
+        for record in records:
+            content = record[0].get("data", {}).get("content", "")
+            message_type = record[0].get("type", "").lower()
+
+            if message_type == "human":
+                messages.append(
+                    HumanMessage(
+                        content=content, additional_kwargs={}, response_metadata={}
+                    )
+                )
+            elif message_type == "ai":
+                messages.append(
+                    AIMessage(
+                        content=content, additional_kwargs={}, response_metadata={}
+                    )
+                )
+            else:
+                print(f"Unknown message type: {message_type}")
+
+        return messages
+
+    @property
+    def messages(self) -> List[BaseMessage]:
+        """Retrieve the messages from FalkorDB for the session.
+
+        Returns:
+            List[BaseMessage]: A list of messages in the current session.
+        """
+        query = (
+            f"MATCH (s:{self._node_label})-[:LAST_MESSAGE]->(last_message) "
+            "WHERE s.id = $session_id MATCH p=(last_message)<-[:NEXT*0.."
```

Comment:
After differentiating graphs by session_id (in self._driver.select_graph(session_id) on line 100), you no longer need to filter by session_id in each query. The graph is unique per session_id, and this applies to all queries.

Reply:
fixed